### ISSUE TITLE
Truncated error messages with 1.95.0

### ISSUE THREAD

**Issue by weiznich** [rust-lang/rust#155414]
<!--
Thank you for filing a regression report! 🐛 A regression is something that changed between versions of Rust but was not supposed to.

Please provide a short summary of the regression, along with any information you feel is relevant to replicate it.
-->

### Code

I tried this code:

```rust
use diesel::prelude::*; // diesel = { version = "2.3.7", default-features = false, features = ["with-deprecated"] }

table! {
    users {
        id -> Integer,
    }
}

#[derive(AsChangeset)]
#[diesel(table_name = users)]
struct UserStruct1 {
    #[column_name = "name"]
    full_name: String,
}

#[derive(AsChangeset)]
#[diesel(table_name = users)]
struct UserStruct2 {
    #[column_name]
    id: i32,
}
```

I expected to see this happen: There is a complete error message containing all wrong instances of wrong attributes in this code snippet.

Example from Rust 1.94.0

```
warning: #[column_name] attribute form is deprecated                                                                                           
  = help: use `#[diesel(column_name = name)]` instead                                                                                          
                                                                                                                                               
error: unexpected end of input, expected `=`                                                                                                   
       help: the correct format looks like `#[diesel(column_name = foo)]`                                                                      
  --> src/main.rs:19:7                                                                                                                         
   |                                                                                                                                           
19 |     #[column_name]                                                                                                                        
   |       ^^^^^^^^^^^                                                                                                                         
                                                                                                                                               
error[E0425]: cannot find type `name` in module `users`                                                                                        
  --> src/main.rs:12:21                                                                                                                        
   |                                                                                                                                           
12 |     #[column_name = "name"]                                                                                                               
   |                     ^^^^^^ not found in `users`                                                                                           
                                                                                                                                               
error[E0425]: cannot find value `name` in module `users`                                                                                       
  --> src/main.rs:12:21                                                                                                                        
   |                                                                                                                                           
12 |     #[column_name = "name"]                                                                                                               
   |                     ^^^^^^ not found in `users`   
```

Instead, this happened: Rustc (1.95.0) only emits a truncated error

```
warning: #[column_name] attribute form is deprecated
  = help: use `#[diesel(column_name = name)]` instead

error: unexpected end of input, expected `=`
       help: the correct format looks like `#[diesel(column_name = foo)]`
  --> src/main.rs:19:7
   |
19 |     #[column_name]
   |       ^^^^^^^^^^^
```

### Version it worked on

<!--
Provide the most recent version this worked on, for example:

It most recently worked on: Rust 1.47
-->

It most recently worked on: 1.94.0

### Version with regression

<!--
Provide the version you are using that has the regression.
-->

`rustc --version --verbose`:
```
rustc 1.95.0 (59807616e 2026-04-14)
binary: rustc
commit-hash: 59807616e1fa2540724bfbac14d7976d7e4a3860
commit-date: 2026-04-14
host: x86_64-unknown-linux-gnu
release: 1.95.0
LLVM version: 22.1.2

```


@rustbot modify labels: +regression-from-stable-to-stable -regression-untriaged

**Comment by weiznich** [comment]
searched nightlies: from nightly-2026-01-17 to nightly-2026-02-27
regressed nightly: nightly-2026-01-19
searched commit range: https://github.com/rust-lang/rust/compare/fe98ddcfcfb6f185dbf4adeaf439d8a756da0273...0a3cd3b6b6e1fa8fd3c75c1d13d2e22e64273f49
regressed commit: https://github.com/rust-lang/rust/commit/2b112efcccf831c0e0fbaa6470047c7067f12801

<details>
<summary>bisected with <a href='https://github.com/rust-lang/cargo-bisect-rustc'>cargo-bisect-rustc</a> v0.6.11</summary>


Host triple: x86_64-unknown-linux-gnu
Reproduce with:
```bash
cargo bisect-rustc --start 1.94.0 --end 1.95.0 --prompt 
```
</details>

Seems to point to https://github.com/rust-lang/rust/pull/149484, which looks reasonable

cc @petrochenkov @ogoffart

@rustbot modify labels: -E-needs-bisection

**Comment by ogoffart** [comment]
Yes, this is #149484 working as intended (from my point of view :-)).

The later errors are suppressed because the first error inside the diesel macro suggests that the missing symbols in that module are likely caused by the macro failure. We want the user to focus on the root cause.

You're right that in this specific case, the later errors are not actually caused by the macro expansion error. So it is not ideal that we don't report all the errors here.

In my opinion, making the root cause more visible and hiding the fallout from the macro error is worth showing fewer errors on the first pass (at least for a human user, maybe for a test suite or an AI agent, the trade-off might go the other way.)

I'll defer to the rust team if they think the change should be reverted.

**Comment by jieyouxu** [comment]
We discussed this diagnostics change during [compiler triage meeting today](https://rust-lang.zulipchat.com/#narrow/channel/238009-t-compiler.2Fmeetings/topic/.5Bweekly.5D.202026-04-23/near/590484734). From team members present during the triage meeting, the thought was:

* The original motivation in #149484 makes sense to help pinpoint the "initial" points of failure, instead of having it drown in a sea of other additional errors. Generally, the tradeoff consideration is that it's better to show less (but the most important) errors instead of reporting all possible errors. However, at times this will unfortunately suppress too many, including genuine ones.
* We want to wait a bit (maybe a release cycle or two) to hear more feedback / examples about this diagnostics change. There were a number of users in the original issue #68838 (that the PR fixed) asking for this behavior so at least some people find it helpful.
  * This is diagnostics only so there's no stability guarantees, we can walk this back based on more feedback if needed.
  * Perhaps it could be tuned, though it's not quite clear in what ways at this moment.
* At this time we don't think a revert is warranted.

We'll tag this as P-medium.
